test/activity_test.rb

```diff
@@ -1079,6 +1079,226 @@ def test(*)
+<*c>
+ {Trailblazer::Activity::Left} => #<End/:failure>
+ {Trailblazer::Activity::Right} => <*d>
+<*d>
+ {Trailblazer::Activity::Right} => <*f>
+<*e>
+ {Trailblazer::Activity::Left} => #<End/:failure>
+ {Trailblazer::Activity::Right} => #<End/:success>
+<*f>
+ {Trailblazer::Activity::Left} => #<End/:failure>
+ {Trailblazer::Activity::Right} => #<End/:success>
+#<End/:success>
+
+#<End/:failure>
+}
+  end
+
+  it "{Path()} allows connecting to the outer step using {Output() => Id()}" do
+    activity = Class.new(Activity::Railway) do
+      step :c, Output(:success) => Path() do
+        step :d, Output(:success) => Id(:f)
+        step :e
+      end
+      step :f
+
+      include T.def_steps(:c, :d, :e, :f)
+    end
+
+    assert_process_for activity, :success, :failure, %{
+#<Start/:default>
+ {Trailblazer::Activity::Right} => <*c>
+<*c>
+ {Trailblazer::Activity::Left} => #<End/:failure>
+ {Trailblazer::Activity::Right} => <*d>
+<*d>
+ {Trailblazer::Activity::Right} => <*f>
+<*e>
+ {Trailblazer::Activity::Right} => #<End/:failure>
+<*f>
+ {Trailblazer::Activity::Left} => #<End/:failure>
+ {Trailblazer::Activity::Right} => #<End/:success>
+#<End/:success>
+
+#<End/:failure>
+}
+
+    signal, (ctx, _) = activity.([{seq: []}])
+
+    signal.inspect.must_equal  %{#<Trailblazer::Activity::End semantic=:success>}
+    ctx.inspect.must_equal     %{{:seq=>[:c, :d, :f]}}
+  end
+
+  it "{Path()} allows emitting signal via {End()}" do
+    activity = Class.new(Activity::Railway) do
+      step :c, Output(:success) => Path() do
+        step :e
+        step :d, Output(:success) => End(:without_cc)
+      end
+      step :f
+
+      include T.def_steps(:c, :e, :d, :f)
+    end
+
+    assert_process_for activity, :success, :without_cc, :success, :failure, %{
+#<Start/:default>
+ {Trailblazer::Activity::Right} => <*c>
+<*c>
+ {Trailblazer::Activity::Left} => #<End/:failure>
+ {Trailblazer::Activity::Right} => <*e>
+<*e>
+ {Trailblazer::Activity::Right} => <*d>
+<*d>
+ {Trailblazer::Activity::Right} => #<End/:without_cc>
+<*f>
+ {Trailblazer::Activity::Left} => #<End/:failure>
+ {Trailblazer::Activity::Right} => #<End/:success>
+#<End/:success>
+
+#<End/:without_cc>
+
+#<End/:failure>
+}
+
+    signal, (ctx, _) = activity.([{seq: []}])
+
+    signal.inspect.must_equal  %{#<Trailblazer::Activity::End semantic=:without_cc>}
+    ctx.inspect.must_equal     %{{:seq=>[:c, :e, :d]}}
+  end
+
+  it "{Path()} allows nesting via {Subprocess()}" do
+    nested = Class.new(Activity::Railway) do
+      step :a
+      step :b, Output(:success) => End(:charge)
+
+      include T.def_steps(:a, :b)
+    end
+
+    activity = Class.new(Activity::Railway) do
+      step :c, Output(:success) => Path() do
+        step :e
+        step Subprocess(nested), Output(:charge) => End(:with_cc)
+        step :d
+      end
+      step :f
+
+      include T.def_steps(:c, :e, :d, :f)
+    end
+
+    assert_process_for activity, :success, :with_cc, :success, :failure, %{
```

Comment:
@apotonick There are 2 `:success` events here because `Path()`'s default `End.success` is also kept including `End.with_cc`.

I am not really sure if it's valid but the reason to keeping is because of `:d`'s output.

Reply:
Two `:success` termini in the same activity? That doesn't sound right? :thinking:

Reply:
Yeah thought so. I'm gonna look at the ways and have it removed correctly.

Reply:
@apotonick Updating the `seq` [here](https://github.com/trailblazer/trailblazer-activity-dsl-linear/pull/21/files#diff-d67c66e8966ebbad9333761b6d61c3d0R52) such that both default `Start.default` and `End.success` ends are removed.

It fixes the scenario of having two `:success` and also removed extra `if` check.